### PR TITLE
switched to http where possible

### LINKED ISSUE #99: Don't use the websocket on load
The websocket isn't available in the first few seconds after page load.
The frontend should use the http endpoints during that time.
The endpoint is https://sky.coflnet.com/command/{type}/base64Data
Responses are cached (don't support logged in commands)
Transmission should switch to websockets after a few seconds or only use it for commands wich require login.

### PR DESCRIPTION
**Pull request by Ekwav**
fixes #99

### CODE REVIEW

File: src/api/HttpHelper.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+function getOrGenerateUUid() : string {
+    let websocketUUID = window.localStorage.getItem("websocketUUID") || generateUUID();
+    window.localStorage.setItem("websocketUUID", websocketUUID);
+    return websocketUUID;
+}
+
+
+export let httpApi:  HttpApi = {
```
Comment (matthias-luger) [inline src/api/HttpHelper.tsx:35]: please use the same formatting style as everywhere else (spaces)

File: src/api/HttpHelper.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    .then(json => request.resolve(json));
+}
+
+function sendRequestLimitCache(request: ApiRequest, grouping = 1): Promise<void> {
+    // invalidate the cache every miniute
+    request.mId = new Date().getMinutes() / grouping;
+    return sendRequest(request);
+}
+
+function getOrGenerateUUid() : string {
```
Comment (matthias-luger) [inline src/api/HttpHelper.tsx:28]: please use the same formatting style as everywhere else (spaces)

File: src/api/HttpHelper.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { ApiRequest, HttpApi, WebsocketHelper } from "./ApiTypes.d";
+import { Base64 } from "js-base64";
+import api from "./ApiHelper";
+import { v4 as generateUUID } from 'uuid';
+
+
+function init(): void {
```
Comment (matthias-luger) [inline src/api/HttpHelper.tsx:7]: remove unused function

File: src/api/HttpHelper.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+}
+
+function sendRequest(request: ApiRequest): Promise<void> {
+    let requestString = JSON.stringify(request.data);
+    let headers = {'ConId':getOrGenerateUUid()};
+    var url = `https://sky.coflnet.com/command/${request.type}/${Base64.encode(requestString)}`;
+    if(request.mId)
```
Comment (matthias-luger) [inline src/api/HttpHelper.tsx:15]: url into constant variable?

**Review by matthias-luger — CHANGES_REQUESTED** [review]
improve according to comments

**Review by matthias-luger — CHANGES_REQUESTED** [review]
improve performance (lighthouse)

**Comment by Ekwav** [comment]
> improve performance (lighthouse)

out of scope

File: public/preScript.js [hunk 5]
```diff
@@ -1,8 +1,9 @@
+/*// not critical anymore
 window.websocket = new WebSocket('wss://skyblock-backend.coflnet.com/skyblock');
 window.websocket.onopen = function() {
     console.log("Websocket opened");
 }
-
+*/
```
Comment (Flou21) [inline public/preScript.js:6]: why?

**Review by Flou21 — APPROVED** [review]

**Pull request merged**